llama_index/tools/retriever_tool.py

```diff
@@ -51,13 +51,23 @@ def retriever(self) -> BaseRetriever:
     def metadata(self) -> ToolMetadata:
         return self._metadata
 
-    def __call__(self, input: Any) -> ToolOutput:
```

Comment:
I wonder if removing this will have any consequences/breaking changes outside of the library? Hopefully we caught all the places that were using this in the library as well 🙏🏻

Reply:
The `AsyncBaseTool` parent class has an implementation of `__call__` which just forwards the invocation to `self.call`. So I think that makes this backwards compatible.

Reply:
Ah right, good point!